lib/puppet/util/firewall.rb

```diff
@@ -160,6 +160,9 @@ def persist_iptables(proto)
 
     # Older iptables-persistent doesn't provide save action.
     if os_key == 'Debian'
+      # We need to call Facter.flush to clear Facter cache as it's possible the cached value will be nil due to the fact
+      # that the iptables-persistent package was potentially installed after the initial Fact gathering.
+      Facter.flush
```

Comment:
Rather than flushing all facts, this could potentially be more specific with `Facter.fact(:iptables_persistent_version).flush`, assuming this fact doesn't depend on cached data that is also out of date.